Assets/Treasure/TDK/Runtime/Services/Thirdweb/TDKThirdwebService.cs

```diff
@@ -134,6 +164,54 @@ public async Task ConnectWallet(EcosystemWalletOptions ecosystemWalletOptions, i
             }
         }
 
+        public async Task ConnectExternalWallet(int chainId)
+        {
+            _connectionCancelationTokenSource?.Cancel();
+            // TODO where to get supported chains list?
+            var supportedChains = new BigInteger[] { chainId, 1 };
+            WalletConnectWallet wallet = await WalletConnectWallet.Create(
+                client: Client,
+                initialChainId: chainId,
+                supportedChains: supportedChains
+            );
+            var options = new EcosystemWalletOptions(authprovider: AuthProvider.Siwe, siweSigner: wallet);
+            await ConnectWallet(options, TDK.Connect.ChainIdNumber, isSilentReconnect: false);
+        }
+
+        // TODO fix heavy cpu load related to unhandled errors appearing when switching chains in metamask after WalletConnect is initialized
```

Comment:
Could be fixed by updating WalletConnect to latest version, waiting on thirdweb for bumping that. I dont think its a blocker for this PR necessarily